pkg/spoke/controllers/manifestwork_controller.go

```diff
@@ -295,7 +295,39 @@ func (m *ManifestWorkController) applyUnstructrued(data []byte, recorder events.
 func (m *ManifestWorkController) generateUpdateStatusFunc(manifestStatus workapiv1.ManifestResourceStatus) helper.UpdateManifestWorkStatusFunc {
 	return func(oldStatus *workapiv1.ManifestWorkStatus) error {
 		oldStatus.ResourceStatus = manifestStatus
-		// TODO aggreagae manifest condition to generate work condition
+		num := len(manifestStatus.Manifests)
+		if num == 0 {
+			oldStatus.Conditions = []workapiv1.StatusCondition{}
+			return nil
+		}
+
+		// aggregate manifest condition to generate work condition
+		applied := 0
+		for _, manifest := range manifestStatus.Manifests {
+			for _, condition := range manifest.Conditions {
+				switch {
+				case condition.Type == string(workapiv1.ManifestApplied) && condition.Status == metav1.ConditionTrue:
+					applied++
```

Comment:
i think it needs comment here indicating how to decide the condition of manifestwork. Maybe a separate func would be better. We can reuse that func for all different condition types.  And having a counter seems not necessary. You just need a bool, and set it false if any condition status is false